ev3dev2/motor.py

```diff
@@ -67,105 +67,126 @@
     raise Exception("Unsupported platform '%s'" % platform)
 
 
-class SpeedInteger(int):
+class SpeedMeasure():
     """
     A base class for other unit types. Don't use this directly; instead, see
     :class:`SpeedPercent`, :class:`SpeedRPS`, :class:`SpeedRPM`,
     :class:`SpeedDPS`, and :class:`SpeedDPM`.
     """
     pass
 
-class SpeedPercent(SpeedInteger):
+class SpeedPercent(SpeedMeasure):
     """
     Speed as a percentage of the motor's maximum rated speed.
     """
 
+    def __init__(self, percent):
+        assert -100 <= percent <= 100,\
+            "{} is an invalid percentage, must be between -100 and 100 (inclusive)".format(percent)
+        
+        self.percent = percent
+
     def __str__(self):
-        return int.__str__(self) + "%"
+        return str(self.percent) + "%"
 
-    def get_speed_pct(self, motor):
+    def to_native_units(self, motor):
         """
-        Return the motor speed percentage represented by this SpeedPercent
+        Return this SpeedPercent in native motor units
         """
-        return self
+        return self.percent / 100 * motor.max_speed
 
 
-class SpeedNativeUnits(SpeedInteger):
+class SpeedNativeUnits(SpeedMeasure):
     """
     Speed in tacho counts per second.
     """
 
+    def __init__(self, native_counts):
+        self.native_counts = native_counts
+    
     def __str__(self):
-        return int.__str__(self) + "% (counts/sec)"
+        return str(self.native_counts) + " (counts/sec)"
 
-    def get_speed_pct(self, motor):
+    def to_native_units(self, motor):
         """
-        Return the motor speed percentage represented by this SpeedNativeUnits
+        Return this SpeedNativeUnits as a number
         """
-        return self/motor.max_speed * 100
+        return self.native_counts
 
-class SpeedRPS(SpeedInteger):
+class SpeedRPS(SpeedMeasure):
     """
     Speed in rotations-per-second.
     """
 
+    def __init__(self, rotations_per_second):
+        self.rotations_per_second = rotations_per_second
+
     def __str__(self):
-        return int.__str__(self) + " rps"
+        return str(self.rotations_per_second) + " rps"
```

Comment:
Here it is 'rps', and above it is 'counts/sec'. Should we stick to one form?

Reply:
I think the string forms should error on the side of clarity over brevity, so how about I change them all to the form `rot/sec`, `deg/min`, etc?

Reply:
That sounds good